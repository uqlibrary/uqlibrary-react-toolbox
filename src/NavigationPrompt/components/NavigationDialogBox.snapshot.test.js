import { mount } from 'enzyme';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import NavigationDialogBox from './NavigationDialogBox';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import Immutable from 'immutable';
import injectTapEventPlugin from 'react-tap-event-plugin';

const create = () => {
    const initialState = Immutable.Map();

    const store = {
        getState: jest.fn(() => (initialState)),
        dispatch: jest.fn(),
        subscribe: jest.fn()
    };
    const next = jest.fn();
    const invoke = (action) => thunk(store)(next)(action);
    return {store, next, invoke}
};

function setup({when, txt, isShallow = true}){
    const props = {
        when: when || true,
        txt: txt || {
            confirmationTitle: 'Confirmation',
            confirmationMessage: 'Are you sure?',
            cancelButtonLabel: 'No',
            confirmButtonLabel: 'Yes'
        }
    };

    if (!isShallow) {
        return mount(
            <Provider store={create().store}>
                <NavigationDialogBox {...props} />
            </Provider>, {
                context: {
                    muiTheme: getMuiTheme()
                },
                childContextTypes: {
                    muiTheme: PropTypes.object.isRequired
                }
            });
    }

    return shallow(<Provider store={create().store}><NavigationDialogBox {...props} /></Provider>);
}

beforeAll(() => {
    injectTapEventPlugin();
});

describe('NavigationDialogBox component', () => {
    it('should render', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
