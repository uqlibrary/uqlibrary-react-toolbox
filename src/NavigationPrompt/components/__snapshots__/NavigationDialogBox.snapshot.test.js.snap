// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationDialogBox component should render 1`] = `
<NavigationDialogBox
  txt={
    Object {
      "cancelButtonLabel": "No",
      "confirmButtonLabel": "Yes",
      "confirmationMessage": "Are you sure?",
      "confirmationTitle": "Confirmation",
    }
  }
  when={true}
/>
`;
